core/src/test/java/org/infinispan/test/transport/DelayedViewJGroupsTransport.java

```diff
@@ -19,11 +24,21 @@ public void receiveClusterView(View newView) {
       // check if this is an event of node going down, and if so wait for a signal to apply new view
       if (waitLatch != null && getMembers().size() > newView.getMembers().size()) {
          try {
-            waitLatch.await(10, TimeUnit.SECONDS);
+            log.debugf("Delaying view %s", newView);
+            unblocked = waitLatch.await(10, TimeUnit.SECONDS);
+            if (unblocked) {
+               log.debugf("Unblocking view %s", newView);
+            } else {
+               log.errorf("Timed out waiting for view to be unblocked: %s", newView);
+            }
          } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
          }
       }
       super.receiveClusterView(newView);
    }
+
+   public void assertUnblocked() {
+      assert unblocked;
```

Comment:
Maybe return boolean and do assertTrue in the calling code?

Reply:
It's not a very strong preference, but I like specific methods because then the error is right there in the stack trace and you don't have to open the source code to see what failed.